src/box/iproto.cc

```diff
@@ -1325,24 +1343,21 @@ iproto_connection_on_output(ev_loop *loop, struct ev_io *watcher,
 {
 	struct iproto_connection *con = (struct iproto_connection *) watcher->data;
 
-	try {
-		int rc;
-		while ((rc = iproto_flush(con)) <= 0) {
-			if (rc != 0) {
-				ev_io_start(loop, &con->output);
-				return;
-			}
-			if (! ev_is_active(&con->input) &&
-			    rlist_empty(&con->in_stop_list)) {
-				ev_feed_event(loop, &con->input, EV_READ);
-			}
+	int rc;
+	while ((rc = iproto_flush(con)) <= 0) {
+		if (rc != 0) {
+			ev_io_start(loop, &con->output);
+			return;
 		}
-		if (ev_is_active(&con->output))
-			ev_io_stop(con->loop, &con->output);
-	} catch (Exception *e) {
-		e->log();
-		iproto_connection_close(con);
+		if (! ev_is_active(&con->input) &&
+		    rlist_empty(&con->in_stop_list)) {
+			ev_feed_event(loop, &con->input, EV_READ);
+		}
+		if (!con->can_write)
+			break;
```

Comment:
This break is redundant. The next iproto_flush() call will return 1, because begin == end. Please remove.

Reply:
No, it will be infinite loop or broken logic without it. Currently we return 0 if `can_write == false`, so loop is infinite. If we return 1 in this case, we not call `ev_feed_event` EV_READ, and connection will never closed.

Reply:
We can move
````
		if (! ev_is_active(&con->input) &&
		    rlist_empty(&con->in_stop_list)) {
			ev_feed_event(loop, &con->input, EV_READ);
		}
```
to iproto_flush and return 1, or we need this break

Reply:
You can move the can_write check after the (begin->used == end->used) check. Then it should work just fine: first we return 0 because of can_write, which will trigger ev_feed_event; on the next iteration of the loop, we return 1, because begin->used == end->used.

Reply:
Fixed, it was child mistake: i write `*end = *begin` instead of `*begin = *end` in `iproto_flush`, that's why it's not work fine